validphys2/src/validphys/overfit_metric.py

```diff
@@ -0,0 +1,128 @@
+"""
+overfit_metric.py
+
+This calculates the overfit metric
+"""
+
+import logging
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import scipy.stats as stats
+
+from reportengine import collect
+from reportengine.figure import figure
+from reportengine.table import table
+
+log = logging.getLogger(__name__)
+
+preds = collect("predictions", ("pdfs","dataset_inputs",))
+
+
+def array_expected_delta_chi2(calculate_chi2s_per_replica, replica_data):
+
+    res_over = calculate_chi2s_per_replica
+
+    fitted_val_erf = np.array([info.validation for info in replica_data])
+
+    number_pdfs = res_over.shape[0]
+    list_expected_delta_chi2 = []
+    for _ in range(number_pdfs * 1000):
+        mask = np.random.randint(0, number_pdfs, size=int(0.95 * number_pdfs))
```

Comment:
If `number_pdfs=1` (I know, it is a patological case). This will return an empty list. Is that expected?

Related to that, maybe the `1000` should instead be an input parameter so that it can be modified more easily.